src/main/java/filescanner/ProgressBarAction.java

```diff
@@ -0,0 +1,22 @@
+package filescanner;
+
+import java.util.concurrent.RecursiveAction;
+
+public class ProgressBarAction extends RecursiveAction {
+    @Override
+    protected void compute() {
+        char[] animationChars = new char[]{'|', '/', '-', '\\'};
+
+        for (int i = 0; i <= 100; i++) {
+            System.out.print("Processing: " + i + "% " + animationChars[i % 4] + "\r");
+
+            try {
+                Thread.sleep(200);
+            } catch (InterruptedException e) {
+                e.printStackTrace();
```

Comment:
such practice is really bad. Please thrown exception, or log it, or wrap to non-checked or use lombok @Sneakythrows(InterruptedException .class) .
Please let me know if you interested on the reason